system-x/services/aws/s3/src/main/java/software/tnb/aws/s3/service/Minio.java

```diff
@@ -0,0 +1,48 @@
+package software.tnb.aws.s3.service;
+
+import software.tnb.aws.common.account.AWSAccount;
+import software.tnb.aws.common.service.AWSService;
+import software.tnb.aws.s3.validation.S3Validation;
+import software.tnb.common.account.Accounts;
+import software.tnb.common.deployment.WithDockerImage;
+
+import java.net.URI;
+
+import software.amazon.awssdk.services.s3.S3Client;
+
+public abstract class Minio extends AWSService<AWSAccount, S3Client, S3Validation> implements WithDockerImage {
+    protected static final int CONTAINER_API_PORT = 9000;
+    protected static final int CONTAINER_UI_PORT = 9001;
+
+    protected abstract String hostname();
```

Comment:
you need to expose the hostname (ideally as the whole `http://hostname:port` string that you can pass to camel component), otherwise you don't know the url when using just `Minio` class (as in tnb-tests). then maybe you won't need the apiPort (will be part of `hostname()`) and uiPort (is not used for testing, if someone wants it, then he can find it manually)